litestar/_openapi/parameters.py

```diff
@@ -206,6 +208,24 @@ def create_parameter_for_handler(
     dependency_providers = route_handler.resolve_dependencies()
     layered_parameters = route_handler.resolve_layered_parameters()
 
+    # if there's a single parameter 'query', surface its fields
+    if "query" in handler_fields and len(handler_fields) == 1:
```

Comment:
I don't want to hold this one up since it's been open for a while, but is `query` the only parameter this applies to?  What about something like `state`?  Does that fall into this same category?